tlatools/org.lamport.tlatools/test/pcal/SBBTETraceTest.java

```diff
@@ -0,0 +1,161 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Microsoft Research. All rights reserved. 
+ *
+ * The MIT License (MIT)
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy 
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software. 
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * Contributors:
+ *   Markus Alexander Kuppe - initial API and implementation
+ ******************************************************************************/
+package pcal;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+
+import tlc2.tool.liveness.ModelCheckerTestCase;
+import tlc2.output.EC;
+import tlc2.output.EC.ExitStatus;
+import tlc2.TraceExpressionSpecTest;
+import tlc2.tool.impl.Tool;
+
+import util.ToolIO;
+
+public class SBBTETraceTest extends TraceExpressionSpecTest {
```

Comment:
@lemmy I'm trying to follow the approach that we use for TE trace testing (see TraceExpressionSpecLassoTest, for example), but when I run `ant -f customBuild.xml test-set -Dtest.testcases="pcal/SBBTETrace*"`, I get

```
    [junit] Failed to match EvaluatingValueTest!A operator override from file:/Users/paulo.feodrippe/dev/tlaplus/tlatools/org.lamport.tlatools/test-class/tlc2/tool/EvaluatingValueTest.class with signature: <Java Method: public static synchronized tlc2.value.impl.Value tlc2.tool.EvaluatingValueTest.action(tlc2.tool.impl.Tool,tla2sany.semantic.ExprOrOpArgNode[],tlc2.util.Context,tlc2.tool.TLCState,tlc2.tool.TLCState,int,tlc2.tool.coverage.CostModel)> (no such module).
    [junit] Warning: Failed to match Get operator override from UserModuleOverrideAnnotation with signature: public static tlc2.value.impl.Value tlc2.overrides.UserModuleOverrideAnnotationImpl.getNumberOne() (no such module).
    [junit] ------------- ---------------- ---------------
    [junit]
    [junit] Testcase: testSpec took 1.049 sec
    [junit] 	Caused an ERROR
    [junit] The invariant of _inv is equal to <Java Method: public static tlc2.value.impl.Value tlc2.module.Sequences.SelectSeq(tlc2.value.impl.Value,tlc2.value.impl.Value)>
    [junit] util.Assert$TLCRuntimeException: The invariant of _inv is equal to <Java Method: public static tlc2.value.impl.Value tlc2.module.Sequences.SelectSeq(tlc2.value.impl.Value,tlc2.value.impl.Value)>
    [junit] 	at util.Assert.fail(Assert.java:33)
    [junit] 	at tlc2.tool.impl.SpecProcessor.processConfigInvariants(SpecProcessor.java:1054)
    [junit] 	at tlc2.tool.impl.SpecProcessor.processConfig(SpecProcessor.java:843)
    [junit] 	at tlc2.tool.impl.SpecProcessor.<init>(SpecProcessor.java:178)
    [junit] 	at tlc2.tool.impl.Spec.<init>(Spec.java:98)
    [junit] 	at tlc2.tool.impl.Tool.<init>(Tool.java:175)
    [junit] 	at tlc2.tool.impl.Tool.<init>(Tool.java:170)
    [junit] 	at tlc2.tool.impl.Tool.<init>(Tool.java:160)
    [junit] 	at tlc2.tool.impl.Tool.<init>(Tool.java:151)
    [junit] 	at tlc2.tool.impl.FastTool.<init>(FastTool.java:47)
    [junit] 	at tlc2.TraceExpressionSpecTest.testSpec(TraceExpressionSpecTest.java:77)
    [junit] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    [junit] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    [junit] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    [junit] 	at util.IsolatedTestCaseRunner.run(IsolatedTestCaseRunner.java:71)
    [junit]
```

I'm not understanding the error `The invariant of _inv is equal...`, is it because it's trying to run TLC twice in the same JVM?

Reply:
I've thought that the `IsolatedTestCaseRunner` would prevent things like that.

Reply:
Do things work correctly in two separate VM runs?

Reply:
When running only the Trace test (I've commented out `tlc.process()` in ...), we don't get this invariant error, it goes until `[junit] ------------- ---------------- ---------------` of the output that I've posted here. It's weird that there is no other output.

But I'll check deeper tomorrow, if it's not possible to run both, I'll just stick to your plan of running them in separate tests.